query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java

```diff
@@ -119,7 +125,11 @@ public boolean isRunning() {
             Collection<Class<?>> javaClasses = (entities.length == 0) ?
                   indexUpdater.allJavaClasses() : Arrays.asList(entities);
             IndexWorker indexWork = new IndexWorker(cache.getName(), javaClasses, skipIndex, null);
-            return executor.timeout(Long.MAX_VALUE, TimeUnit.SECONDS).submitConsumer(indexWork, TRI_CONSUMER);
+            ClusterExecutor clusterExecutor = executor.timeout(Long.MAX_VALUE, TimeUnit.SECONDS);
+            if (localOnly) {
+               clusterExecutor = clusterExecutor.filterTargets(a -> a.equals(cache.getRpcManager().getAddress()));
```

Comment:
I'm 99% sure that `cache.getRpcManager()` returns `null` for local & simple caches

Reply:
yes, but I got lucky here since the ```ClusterExecutor``` for a local cache does not execute the predicate :smile: